packages/lib/src/application/App.ts

```diff
@@ -37,27 +37,27 @@ export default class App {
 
   constructor(appContract: ContractWrapper, txParams: any = {}) {
     this.appContract = appContract;
-    this.txParams = txParams;
+    this.txParams = { ...txParams };
   }
 
   public async getPackage(name): Promise<{ package: Package, version: string }> {
-    const [address, version] = await this.appContract.getPackage(name);
-    const thepackage = await Package.fetch(address, this.txParams);
+    const { ['0']: address, ['1']: version } = await this.appContract.methods.getPackage(name).call();
```

Comment:
I'm afraid to ask why this change was needed. How are tuples returned now in web3 1.0?!

Reply:
If a Solidity function returns named tuples, like
(Solidity) `function myFunction() public view returns (address adr1, address adr2)`
You can do
(Javascript) `const { adr1, adr2 } = await contract.methods.myFunction().call()`

But if the solidity function does not name the members of the tuple:
(Solidity) `function myFunction() public view returns (address, address)`
You can do
(Javascript) `const { ['0']: adr1, ['1']: adr2 } = await contract.methods.myFunction().call()`

Since Web3 v0 exposed tuples as arrays, this use to work:
(Javascript) `const [ adr1, adr2 ] = await contract.myFunction()`